Upstream-Status: Inappropriate [libmali specific]

Index: mesa-demos-8.5.0/src/egl/opengl/eglinfo.c
===================================================================
--- mesa-demos-8.5.0.orig/src/egl/opengl/eglinfo.c
+++ mesa-demos-8.5.0/src/egl/opengl/eglinfo.c
@@ -94,12 +94,14 @@
       .human_name = "X11",
       .platform_enum = EGL_PLATFORM_X11_KHR,
    },
+#ifdef EGL_PLATFORM_SURFACELESS_MESA
    {
       .names = { "EGL_MESA_platform_surfaceless" },
       .short_name = "surfaceless",
       .human_name = "Surfaceless",
       .platform_enum = EGL_PLATFORM_SURFACELESS_MESA,
    },
+#endif
 };
 
 struct eglconfig_info {
@@ -330,6 +332,7 @@
 }
 
 
+#ifdef EGL_EXT_device_base
 static const char *
 PrintDeviceExtensions(EGLDeviceEXT d, EGLBoolean single_line)
 {
@@ -347,6 +350,7 @@
    print_extension_list(extensions, single_line);
    return extensions;
 }
+#endif
 
 
 static const char*
@@ -632,6 +636,7 @@
 }
 
 
+#ifdef EGL_EXT_device_base
 static int
 doOneDevice(EGLDeviceEXT d, int i, struct options opts)
 {
@@ -651,8 +656,10 @@
    return 0;
 #endif
 }
+#endif
 
 
+#ifdef EGL_EXT_device_base
 static int
 doDevices(const char *name, struct options opts)
 {
@@ -681,6 +688,7 @@
 
    return ret;
 }
+#endif
 
 
 static void
@@ -857,10 +865,12 @@
          }
       }
 
+#ifdef EGL_EXT_device_base
       if (strstr(clientext, "EGL_EXT_device_enumeration") &&
           strstr(clientext, "EGL_EXT_platform_device") &&
           opts.platform == ALL)
          ret += doDevices("Device platform", opts);
+#endif
    }
    else {
       ret = doOneDisplay(eglGetDisplay(EGL_DEFAULT_DISPLAY), "Default display", opts);
