From 0ac93620b3c14a2cb109c24048dd562a0a456fd3 Mon Sep 17 00:00:00 2001
From: Mahesh Bodapati <mbodapat@xilinx.com>
Date: Fri, 29 Sep 2017 18:00:23 +0530
Subject: [PATCH 2/6] fixed bug in GCC so that It will support .long 0U and
 .long 0u

Signed-off-by: Aayush Misra <aayushm@amd.com>

Upstream-Status: Pending
---
 gas/expr.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gas/expr.c b/gas/expr.c
index 87824b0ca1a..15f1b452371 100644
--- a/gas/expr.c
+++ b/gas/expr.c
@@ -861,6 +861,15 @@ operand (expressionS *expressionP, enum expr_mode mode)
 	      break;
 	    }
 	}
+      if ((*input_line_pointer == 'U') || (*input_line_pointer == 'u'))
+      {
+      input_line_pointer--;
+
+      integer_constant ((NUMBERS_WITH_SUFFIX || flag_m68k_mri)
+                        ? 0 : 10,
+                        expressionP);
+      break;
+      }
       c = *input_line_pointer;
       switch (c)
 	{
-- 
2.47.3


