DRI: Add xlnx dri

Add the Xilinx dri target

Upstream-Status: Pending

Signed-off-by: Mark Hatle <mark.hatle@amd.com>

Index: mesa-24.0.1/src/gallium/targets/dri/meson.build
===================================================================
--- mesa-24.0.1.orig/src/gallium/targets/dri/meson.build
+++ mesa-24.0.1/src/gallium/targets/dri/meson.build
@@ -101,6 +101,7 @@ foreach d : [[with_gallium_kmsro, [
                'stm_dri.so',
                'sun4i-drm_dri.so',
                'udl_dri.so',
+               'xlnx_dri.so',
              ]],
              [with_gallium_radeonsi, 'radeonsi_dri.so'],
              [with_gallium_nouveau, 'nouveau_dri.so'],
Index: mesa-24.0.1/src/gallium/targets/dri/target.c
===================================================================
--- mesa-24.0.1.orig/src/gallium/targets/dri/target.c
+++ mesa-24.0.1/src/gallium/targets/dri/target.c
@@ -129,6 +129,7 @@ DEFINE_LOADER_DRM_ENTRYPOINT(sti)
 DEFINE_LOADER_DRM_ENTRYPOINT(stm)
 DEFINE_LOADER_DRM_ENTRYPOINT(sun4i_drm)
 DEFINE_LOADER_DRM_ENTRYPOINT(udl)
+DEFINE_LOADER_DRM_ENTRYPOINT(xlnx)
 #endif
 
 #if defined(GALLIUM_LIMA)
