Upstream-Status: Not-submitted [Needs verification]
---
diff a/fdisks/fdiskbsdlabel.h b/fdisks/fdiskbsdlabel.h
--- a/fdisks/fdiskbsdlabel.h
+++ b/fdisks/fdiskbsdlabel.h
@@ -55,7 +55,7 @@
 #elif defined (__alpha__) || defined (__powerpc__) || defined (__ia64__) || defined (__hppa__)
 #define BSD_LABELSECTOR   0
 #define BSD_LABELOFFSET   64
-#elif defined (__s390__) || defined (__s390x__)
+#elif defined (__s390__) || defined (__s390x__) || defined (__microblaze__) || defined (__microblazeel__)
 #define BSD_LABELSECTOR   1
 #define BSD_LABELOFFSET   0
 #else
