From 53b76bb548720367032a51a6d604c975b10bb30e Mon Sep 17 00:00:00 2001
From: Aayush Misra <aayushm@amd.com>
Date: Fri, 29 Mar 2024 14:59:16 +0530
Subject: [PATCH] gdb/gdserver: Fix ABI settings for gdbserver

---
 gdb/microblaze-tdep.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/gdb/microblaze-tdep.c b/gdb/microblaze-tdep.c
index 38ba38e8c7d..35cec286d8f 100644
--- a/gdb/microblaze-tdep.c
+++ b/gdb/microblaze-tdep.c
@@ -1120,12 +1120,13 @@ void _initialize_microblaze_tdep ();
 void
 _initialize_microblaze_tdep ()
 {
+  //Setting abi to auto manually, should be able to modify in 'arch'_gdbarch_init function
+  microblaze_abi_string = microblaze_abi_strings[0]; 
+
   register_gdbarch_init (bfd_arch_microblaze, microblaze_gdbarch_init);
-// static struct cmd_list_element *setmicroblazecmdlist = NULL;
-// static struct cmd_list_element *showmicroblazecmdlist = NULL;
 
-  /* Add root prefix command for all "set microblaze"/"show microblaze" commands.  */
 
+  /* Add root prefix command for all "set microblaze"/"show microblaze" commands.  */
   add_setshow_prefix_cmd ("microblaze", no_class,
 			 _("Various microblaze specific commands."),
 			 _("Various microblaze specific commands."),
-- 
2.34.1

