diff -Naur dpkg-1.15.8.7/m4/dpkg-compiler.m4 tmp/m4/dpkg-compiler.m4
--- dpkg-1.15.8.7/m4/dpkg-compiler.m4	2010-12-19 17:25:34.000000000 -0800
+++ tmp/m4/dpkg-compiler.m4	2013-03-11 22:27:10.000000000 -0700
@@ -13,7 +13,7 @@
 
 WFLAGS="-Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers \
 	 -Wmissing-declarations -Wmissing-format-attribute \
-	 -Wvla -Winit-self -Wwrite-strings -Wcast-align -Wshadow"
+	 -Winit-self -Wwrite-strings -Wcast-align -Wshadow"
 WCFLAGS="-Wdeclaration-after-statement -Wnested-externs -Wbad-function-cast \
 	 -Wstrict-prototypes -Wmissing-prototypes -Wold-style-definition"
 # Temporarily here until #542031 gets fixed in ncurses
