Add "microblazeel" target to config.sub.

Index: ncurses/config.sub
===================================================================
--- ncurses.orig/config.sub	2011-10-17 19:49:28.000000000 +1000
+++ ncurses/config.sub	2012-10-04 16:43:53.250376336 +1000
@@ -154,7 +154,7 @@
 	-convergent* | -ncr* | -news | -32* | -3600* | -3100* | -hitachi* |\
 	-c[123]* | -convex* | -sun | -crds | -omron* | -dg | -ultra | -tti* | \
 	-harris | -dolphin | -highlevel | -gould | -cbm | -ns | -masscomp | \
-	-apple | -axis | -knuth | -cray | -microblaze)
+	-apple | -axis | -knuth | -cray | -microblaze | -microblazeel)
 		os=
 		basic_machine=$1
 		;;
@@ -260,7 +260,7 @@
 	| ip2k | iq2000 \
 	| lm32 \
 	| m32c | m32r | m32rle | m68000 | m68k | m88k \
-	| maxq | mb | microblaze | mcore | mep | metag \
+	| maxq | mb | microblaze | microblazeel | mcore | mep | metag \
 	| mips | mipsbe | mipseb | mipsel | mipsle \
 	| mips16 \
 	| mips64 | mips64el \
@@ -357,7 +357,7 @@
 	| lm32-* \
 	| m32c-* | m32r-* | m32rle-* \
 	| m68000-* | m680[012346]0-* | m68360-* | m683?2-* | m68k-* \
-	| m88110-* | m88k-* | maxq-* | mcore-* | metag-* | microblaze-* \
+	| m88110-* | m88k-* | maxq-* | mcore-* | metag-* | microblaze-* | microblazeel-* \
 	| mips-* | mipsbe-* | mipseb-* | mipsel-* | mipsle-* \
 	| mips16-* \
 	| mips64-* | mips64el-* \
@@ -758,6 +758,9 @@
 	microblaze)
 		basic_machine=microblaze-xilinx
 		;;
+	microblaze)
+		basic_machine=microblazeel-xilinx
+		;;
 	mingw32)
 		basic_machine=i386-pc
 		os=-mingw32
