From: "Edgar E. Iglesias" <edgar@axis.com>
Date: Mon, 27 Feb 2012 06:28:00 +0100
Subject: Robustify inline function support

Signed-off-by: Edgar E. Iglesias <edgar@axis.com>
Upstream-Status: Pending
---
 gdb/blockframe.c   |    3 ++-
 gdb/inline-frame.c |    3 +++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/gdb/blockframe.c b/gdb/blockframe.c
index d5787f1..80e40a3 100644
--- a/gdb/blockframe.c
+++ b/gdb/blockframe.c
@@ -78,7 +78,8 @@ get_frame_block (struct frame_info *frame, CORE_ADDR *addr_in_block)
 	inline_count--;
 
       bl = BLOCK_SUPERBLOCK (bl);
-      gdb_assert (bl != NULL);
+      if (!bl)
+        return NULL;
     }
 
   return bl;
diff --git a/gdb/inline-frame.c b/gdb/inline-frame.c
index 2773814..c3a77ad 100644
--- a/gdb/inline-frame.c
+++ b/gdb/inline-frame.c
@@ -230,6 +230,9 @@ inline_frame_sniffer (const struct frame_unwind *self,
       cur_block = BLOCK_SUPERBLOCK (cur_block);
     }
 
+  if (depth == 0)
+    return 0;
+
   /* Check how many inlined functions already have frames.  */
   for (next_frame = get_next_frame (this_frame);
        next_frame && get_frame_type (next_frame) == INLINE_FRAME;
-- 
1.7.9.5

