From 4ebe009d505bf10b0d90bad081d3fe5bd1cf7441 Mon Sep 17 00:00:00 2001
From: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
Date: Thu, 6 May 2021 14:33:42 -0700
Subject: [PATCH] Add enable/disable libudev

Upstream-Status: Pending
Signed-off-by: Jeremy Puhlman <jpuhlman@mvista.com>
Signed-off-by: Sai Hari Chandana Kalluri <chandana.kalluri@xilinx.com>
---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index c2ef164..612f987 100755
--- a/configure
+++ b/configure
@@ -1633,6 +1633,10 @@ for opt do
   ;;
   --gdb=*) gdb_bin="$optarg"
   ;;
+  --enable-libudev) libudev="yes"
+  ;;
+  --disable-libudev) libudev="no"
+  ;;
   --enable-rng-none) rng_none=yes
   ;;
   --disable-rng-none) rng_none=no
-- 
2.7.4

