Subject: MicroBlaze Support

Upstream-Status: Not Submitted
---
diff --git a/src/libaio.h b/src/libaio.h
--- a/src/libaio.h
+++ b/src/libaio.h
@@ -117,6 +117,16 @@
 #define PADDEDptr(x, y) x
 #define PADDEDul(x, y)  unsigned long x
 #  endif
+#elif defined(__microblaze__)
+#  if defined (__BIG_ENDIAN__) /* big endian, 32 bits */
+#define PADDED(x, y)   unsigned y; x
+#define PADDEDptr(x, y)        unsigned y; x
+#define PADDEDul(x, y) unsigned y; unsigned long x
+#  else /* little endian, 32 bits */
+#define PADDED(x, y)   x; unsigned y
+#define PADDEDptr(x, y)        x; unsigned y
+#define PADDEDul(x, y) unsigned long x; unsigned y
+#  endif
 #else
 #error endian?
 #endif
---
