From c7e7aea9c60f87096a38604a9d4738cf762669dd Mon Sep 17 00:00:00 2001
From: Aayush Misra <aayushm@amd.com>
Date: Fri, 26 Jul 2024 19:08:17 +0530
Subject: [PATCH 11/11] Use port-specific sbrk

---
 libgloss/microblaze/Makefile.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libgloss/microblaze/Makefile.in b/libgloss/microblaze/Makefile.in
index 6d369242d..d5206a849 100644
--- a/libgloss/microblaze/Makefile.in
+++ b/libgloss/microblaze/Makefile.in
@@ -122,6 +122,7 @@ linux-crt0.o: linux-crt0.S
 crtinit.o: crtinit.S
 sim-crtinit.o: sim-crtinit.S
 sim-pgcrtinit.o: sim-pgcrtinit.S
+sbrk.o: sbrk.c
 
 # target specific makefile fragment comes in here.
 @target_makefile_frag@
-- 
2.34.1

